homework/src/index.js

```diff
@@ -30,18 +45,78 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+  function renderRepositoryInfo(info) {
+    const li = createAndAppend('li', ul);
+    createAndAppend('span', li, { id: 'title', text: 'Repository' });
+    const p1 = createAndAppend('p', li);
+    createAndAppend('a', p1, { href: info.RepositoryUrl, target: '_blank', text: info.Repository });
+    const li2 = createAndAppend('li', ul);
+    createAndAppend('span', li2, { id: 'title', text: 'Description:  ' });
+    createAndAppend('p', li2, { text: info.Description });
+    const li3 = createAndAppend('li', ul);
+    createAndAppend('span', li3, { id: 'title', text: 'Forks: ' });
+    createAndAppend('p', li3, { text: info.Forks });
+    const li4 = createAndAppend('li', ul);
+    createAndAppend('span', li4, { id: 'title', text: 'Updated: ' });
+    createAndAppend('p', li4, { text: info.Updated });
+  }
+
+  async function renderContributorsInfo(info) {
+    try {
+      const contributors = await fetchJSON(info.ContributorsUrl);
+      ul2.innerText = '';
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul2);
+        createAndAppend('img', li, { src: contributor.avatar_url });
+        createAndAppend('a', li, { href: contributor.html_url, target: '_blank', text: contributor.login });
+        createAndAppend('span', li, { text: contributor.contributions });
+      });
+    }
+    catch (err) {
+      createAndAppend('div', contributionsInfo, { text: err.message, class: 'alert-error' });
+    }
+  }
+
+  async function main(url) {
+    try {
+      const repositories = await fetchJSON(url);
+      const select = createAndAppend('select', header);
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      const sortedQueries = [];
+      repositories.forEach(repository => {
+        const item = { Repository: repository.name, Description: repository.description, Forks: repository.forks, Updated: repository.updated_at, ContributorsUrl: repository.contributors_url, RepositoryUrl: repository.html_url };
+        sortedQueries.push(item);
+      });
```

Comment:
This is a comment I already made in week 2:

>There is no need to create a separate array (here: `sortedQueries`), you could use the repositories array instead. Your `sortedQueries` array is just a subset of fields of the repositories array. Nothing is gained from creating this extra array, it is just adding 'noise'.

Note also that object property names should follow the naming conventions for JavaScript variable names: camelCase.